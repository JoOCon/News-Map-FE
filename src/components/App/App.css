.toggle-map-style,
.toggle-map-style-active {
  height: 1.6rem;
  width: 3.4rem;
  border-radius: 1rem;
  margin: auto;
  border: 2px solid #afafaf;
  text-decoration-color: none; }

button:focus {
  outline: 0; }

.toggle-map-style {
  background-color: #eeeeee; }

.toggle-map-style-active {
  background-color: #3c3939; }

.streets-button:hover {
  background-color: gray; }

.streets-button,
.dark-button {
  height: 1.55rem;
  width: 1.6rem;
  border-radius: 50%;
  align-self: start;
  cursor: pointer; }

.dark-button {
  background-color: gray;
  float: right;
  cursor: pointer; }

.dark-button:hover {
  background-color: whitesmoke; }
