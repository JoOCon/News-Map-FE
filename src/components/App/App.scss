.login-button,
.logout-button {
}

.toggle-map-style,
.toggle-map-style-active {
  height: 1.6rem;
  width: 3.4rem;
  border-radius: 1rem;
  margin: auto;
  border: 2px solid rgb(175, 175, 175);
  text-decoration-color: none;
}

button:focus {
  outline: 0;
}

.toggle-map-style {
  background-color: rgb(238, 238, 238);
}

.toggle-map-style-active {
  background-color: rgb(60, 57, 57);
}

.streets-button:hover {
  background-color: gray;
}

.streets-button,
.dark-button {
  height: 1.55rem;
  width: 1.6rem;
  border-radius: 50%;
  align-self: start;
  cursor: pointer;
}

.dark-button {
  background-color: gray;
  float: right;
  cursor: pointer;
}

.dark-button:hover {
  background-color: whitesmoke;
}
