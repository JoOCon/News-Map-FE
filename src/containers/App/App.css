.App {
  background-color: #646464;
  background-image: url("https://newevolutiondesigns.com/images/freebies/city-wallpaper-5.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 55em;
  font-family: 'Roboto Slab', serif; }

.placement-container {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center; }

.login-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(200, 90, 0, 0.9);
  padding: .3em 1em;
  border-radius: .3em;
  box-shadow: 0.1em 0.2em #646464; }

.login-container h1 {
  margin: 0;
  font-size: 3em;
  color: white; }

.login-button {
  font-family: 'Roboto Slab', serif;
  margin: .5em;
  padding: .2em .5em;
  background-color: #646464;
  border: 1px solid transparent;
  border-radius: .3em;
  color: white;
  font-size: 2em;
  cursor: pointer; }

.login-button:hover {
  border: 1px solid #ffd732;
  transition: 500ms; }

.toggle-map-style,
.toggle-map-style-active {
  height: 1.6rem;
  width: 3.4rem;
  border-radius: 1rem;
  margin: auto;
  border: 2px solid #afafaf;
  text-decoration-color: none; }

button:focus {
  outline: 0; }

.toggle-map-style {
  background-color: #eeeeee; }

.toggle-map-style-active {
  background-color: #3c3939; }

.streets-button:hover {
  background-color: gray; }

.streets-button,
.dark-button {
  height: 1.55rem;
  width: 1.6rem;
  border-radius: 50%;
  align-self: start;
  cursor: pointer; }

.dark-button {
  background-color: gray;
  float: right;
  cursor: pointer; }

.dark-button:hover {
  background-color: whitesmoke; }

.main-container {
  display: flex; }
