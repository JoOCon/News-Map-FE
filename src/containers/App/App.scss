.App {
  background-color: rgb(100, 100, 100);
  background-image: url('https://newevolutiondesigns.com/images/freebies/city-wallpaper-5.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 55em;
  font-family: 'Roboto Slab', serif;
}

.placement-container {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.login-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(200, 90, 0, 0.9);
  padding: 0.3em 1em;
  border-radius: 0.3em;
  box-shadow: 0.1em 0.2em rgba(100, 100, 100, 1);
}

.login-container h1 {
  margin: 0;
  font-size: 3em;
  color: rgb(255, 255, 255);
}

.login-button {
  font-family: 'Roboto Slab', serif;
  margin: 0.5em;
  padding: 0.2em 0.5em;
  background-color: #646464;
  border: 1px solid transparent;
  border-radius: 0.3em;
  color: rgb(255, 255, 255);
  font-size: 2em;
  cursor: pointer;
}

.login-button:hover {
  border: 1px solid rgba(255, 215, 50, 1);
  transition: 500ms;
}

.login-button,
.logout-button {
}

.toggle-map-style,
.toggle-map-style-active {
  height: 1.6rem;
  width: 3.4rem;
  border-radius: 1rem;
  margin: auto;
  border: 2px solid rgb(175, 175, 175);
  text-decoration-color: none;
}

button:focus {
  outline: 0;
}

.toggle-map-style {
  background-color: rgb(238, 238, 238);
  position: absolute;
  z-index: 10;
  left: 50%;
  top: 1.5%;
}

.toggle-map-style-active {
  background-color: #5d5a5a;
  position: absolute;
  z-index: 10;
  left: 50%;
  top: 1.5%;
}

.streets-button:hover {
  background-color: gray;
}

.streets-button,
.dark-button {
  height: 1.55rem;
  width: 1.6rem;
  border-radius: 50%;
  cursor: pointer;
}

.dark-button {
  background-color: gray;
  float: right;
  cursor: pointer;
}

.dark-button:hover {
  background-color: whitesmoke;
}

.main-container {
  display: flex;
}

.fa-window-close {
  position: absolute;
  right: 9.5%;
  top: 1.3%;
  font-size: 2rem;
  z-index: 30;
  cursor: pointer;
  padding: 0.3rem;
}

.fa-window-close:hover {
  border-radius: 50%;
  animation: dark-pulse 1.3s linear infinite;
}

.fa-bars {
  position: absolute;
  left: 98%;
  top: 1.3%;
  font-size: 2rem;
  z-index: 30;
  border-radius: 50%;
  padding: 0.5rem;
  margin: 0 0 0 -2rem;
  cursor: pointer;
  color: rgb(117, 72, 72);
}

.dark {
  color: white;
}

@keyframes light-pulse {
  0% {
    box-shadow: 0 0 0 0 rgba(228, 228, 228, 0.226);
  }
  50% {
    box-shadow: 0 0 0 7px rgba(207, 206, 206, 0.151);
  }
  100% {
    box-shadow: 0 0 0 0 rgba(251, 255, 255, 0.425);
  }
}

@keyframes dark-pulse {
  0% {
    box-shadow: 0 0 0 0 rgba(252, 250, 250, 0);
  }
  50% {
    box-shadow: 0 0 0 7px rgba(15, 15, 15, 0.151);
  }
  100% {
    box-shadow: 0 0 0 0 rgba(251, 255, 255, 0.082);
  }
}
