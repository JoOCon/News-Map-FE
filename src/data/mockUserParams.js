export const responseGoogle1 =
  'https://event-mapper-api.herokuapp.com/api/v1/users';

export const responseGoogle2 = {
  body:
    '{"user":{"google_id":"117977763637801404146","given_name":"Paul","family_name":"K","email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const getEvents =
  'https://app.ticketmaster.com/discovery/v2/events.json?apikey=GddBLHQGZLJjPNWsNxOrYRAoBx8Hdqrf&geoPoint=9wjr4et3f&radius=200&size=100';

export const postUser = {
  body: '{"user":{"email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const setFavorite = {
  body: '{}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const removeEvent =
  'https://event-mapper-api.herokuapp.com/api/v1/users/undefined/events/undefined';

export const removeEventOptions = {
  headers: { 'Content-Type': 'application/json' },
  method: 'DELETE'
};
