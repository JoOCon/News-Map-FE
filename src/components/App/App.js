import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import { GoogleLogin, GoogleLogout } from 'react-google-login';

import * as keys from '../../utilities/apiCalls/apiKeys';
import Map from '../Map/Map';
import { postUser } from '../../utilities/apiCalls/apiCalls';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      user: null,
      mapType: 'streets'
    };
  }

  responseGoogle = res => {
    postUser(res.profileObj);
    this.setState({ user: res });
  };

  logout = res => {
    this.setState({ user: null });
  };

  render() {
    const { user } = this.state;
    return (
      <div className="App">
        {!user && (
          <GoogleLogin
            clientId={keys.googleClientId}
            buttonText="Login w/ Google"
            onSuccess={this.responseGoogle}
            onFailure={this.responseGoogle}
          />
        )}
        {user && (
          <div>
            <GoogleLogout buttonText="Logout" onLogoutSuccess={this.logout} />
            <Map />
          </div>
        )}
        <header className="App-header">
          <h1>Hello World</h1>
        </header>
      </div>
    );
  }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
