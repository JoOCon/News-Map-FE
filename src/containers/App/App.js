import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { loginUser, setUserLocation } from '../../actions';
import { Routes } from '../../components/Routes/Routes';
import NavBar from '../../components/NavBar/NavBar';

import { postUser } from '../../utilities/apiCalls/apiCalls';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      user: null,
      mapType: 'streets',
      displaySidebar: false,
      redirect: false
    };
  }

  async componentDidMount() {
    await this.setLatLngEvents();
  }

  setLatLngEvents = () => {
    navigator.geolocation.getCurrentPosition(location => {
      const { latitude, longitude } = location.coords;
      this.props.setUserLocation({ latitude, longitude });
    });
  };

  login = async res => {
    if (!res.email) {
      console.log(res);
      return;
    }
    const activeUser = await postUser(res.profileObj);
    this.props.loginUser(activeUser);
    this.setState({ user: activeUser, redirect: true });
  };

  logout = () => {
    this.setState({ user: null, redirect: false });
  };

  changeMap = (event, style) => {
    event.preventDefault();
    const { mapType } = this.state;
    mapType !== style
      ? this.setState({ mapType: style })
      : this.setState({ mapType: 'streets' });
  };

  displaySidebar = () => {
    this.setState({ displaySidebar: !this.state.displaySidebar });
  };

  render() {
    const { mapType, redirect, displaySidebar, user } = this.state;
    return (
      <Router>
        <div>
          {user && (
            <div>
              <div
                className={
                  mapType === 'streets'
                    ? 'toggle-map-style'
                    : 'toggle-map-style-active'
                }
              >
                <button
                  className={`${mapType}-button`}
                  onClick={event => this.changeMap(event, 'dark')}
                />
              </div>
              <i
                className={
                  !displaySidebar
                    ? `fas fa-bars ${mapType}`
                    : 'far fa-window-close'
                }
                onClick={this.displaySidebar}
              />
            </div>
          )}
          {displaySidebar && <NavBar logout={this.logout} />}
          <Routes
            redirect={redirect}
            changeMap={this.changeMap}
            displaySidebar={this.displaySidebar}
            stateSidebar={displaySidebar}
            user={user}
            mapType={mapType}
            login={this.login}
          />
        </div>
      </Router>
    );
  }
}

App.propTypes = {
  user: PropTypes.object,
  loginUser: PropTypes.func,
  setUserLocation: PropTypes.func
};

export const mapDispatchToProps = dispatch => ({
  loginUser: user => dispatch(loginUser(user)),
  setUserLocation: coordinates => dispatch(setUserLocation(coordinates))
});

export default connect( null, mapDispatchToProps)(App);
