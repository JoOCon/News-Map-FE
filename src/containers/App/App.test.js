import React from 'react';

import { App, mapDispatchToProps } from './App';
import * as user from '../../data/mockUser';
import * as params from '../../data/mockUserParams';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({})
}));

describe('App', () => {
  let wrapper;
  let mockLogin = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App loginUser={mockLogin} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when user is logged in', () => {
    wrapper.setState({ user: { name: 'paul' } });
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when displaySidebar is true', () => {
    wrapper.setState({ displaySidebar: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.loginUser();

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call postUser with correct Params', async () => {
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(user.mockStateUser)
      })
    );

    await wrapper.instance().login(user.mockUser);

    expect(window.fetch).toHaveBeenCalledWith(
      params.responseGoogle1,
      params.responseGoogle2
    );
    expect(mockLogin).toHaveBeenCalled();
    expect(wrapper.state().user).toEqual(user.mockStateUser);
  });

  it('should change state of user when logut is called', () => {
    wrapper.setState({ user: user.mockUser.profileObj });

    wrapper.instance().logout();

    expect(wrapper.state().user).toEqual(null);
  });

  it('should setState of mapType when changeMap is called', () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().changeMap(mockEvent, 'dark');

    expect(wrapper.state().mapType).toEqual('dark');

    wrapper.instance().changeMap(mockEvent, 'streets');

    expect(wrapper.state().mapType).toEqual('streets');
  });

  it('should setState when displaySideBar is called', () => {
    expect(wrapper.state().displaySidebar).toEqual(false);

    wrapper.instance().displaySidebar();

    expect(wrapper.state().displaySidebar).toEqual(true);
  });

  it('should setState of displaySidebar on icon click', () => {
    expect(wrapper.state().displaySidebar).toEqual(false);

    wrapper.setState({ user: user.mockUser.profileObj });
    wrapper.update();
    wrapper.find('i').simulate('click');

    expect(wrapper.state().displaySidebar).toEqual(true);
  });

  it('should setState of mapType on button click', () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.setState({ user: user.mockUser.profileObj });

    expect(wrapper.state().mapType).toEqual('streets');

    wrapper.find('button').simulate('click', mockEvent, 'dark');

    expect(wrapper.state().mapType).toEqual('dark');
  });
});
