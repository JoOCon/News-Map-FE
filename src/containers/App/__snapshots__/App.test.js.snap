// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loginUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Routes
          changeMap={[Function]}
          displaySidebar={[Function]}
          login={[Function]}
          mapType="streets"
          redirect={false}
          stateSidebar={false}
          user={null}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          false,
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={false}
            user={null}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeMap": [Function],
            "displaySidebar": [Function],
            "login": [Function],
            "mapType": "streets",
            "redirect": false,
            "stateSidebar": false,
            "user": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={false}
            user={null}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            false,
            <Routes
              changeMap={[Function]}
              displaySidebar={[Function]}
              login={[Function]}
              mapType="streets"
              redirect={false}
              stateSidebar={false}
              user={null}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          null,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeMap": [Function],
              "displaySidebar": [Function],
              "login": [Function],
              "mapType": "streets",
              "redirect": false,
              "stateSidebar": false,
              "user": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match snapshot when displaySidebar is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loginUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Connect(NavBar)
          logout={[Function]}
        />
        <Routes
          changeMap={[Function]}
          displaySidebar={[Function]}
          login={[Function]}
          mapType="streets"
          redirect={false}
          stateSidebar={true}
          user={null}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Connect(NavBar)
            logout={[Function]}
          />,
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={true}
            user={null}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "logout": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeMap": [Function],
            "displaySidebar": [Function],
            "login": [Function],
            "mapType": "streets",
            "redirect": false,
            "stateSidebar": true,
            "user": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Connect(NavBar)
            logout={[Function]}
          />
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={true}
            user={null}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <Connect(NavBar)
              logout={[Function]}
            />,
            <Routes
              changeMap={[Function]}
              displaySidebar={[Function]}
              login={[Function]}
              mapType="streets"
              redirect={false}
              stateSidebar={true}
              user={null}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "logout": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeMap": [Function],
              "displaySidebar": [Function],
              "login": [Function],
              "mapType": "streets",
              "redirect": false,
              "stateSidebar": true,
              "user": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match snapshot when user is logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loginUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <div>
          <div
            className="toggle-map-style"
          >
            <button
              className="streets-button"
              onClick={[Function]}
            />
          </div>
          <i
            className="fas fa-bars streets"
            onClick={[Function]}
          />
        </div>
        <Routes
          changeMap={[Function]}
          displaySidebar={[Function]}
          login={[Function]}
          mapType="streets"
          redirect={false}
          stateSidebar={false}
          user={
            Object {
              "name": "paul",
            }
          }
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="toggle-map-style"
            >
              <button
                className="streets-button"
                onClick={[Function]}
              />
            </div>
            <i
              className="fas fa-bars streets"
              onClick={[Function]}
            />
          </div>,
          false,
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={false}
            user={
              Object {
                "name": "paul",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="toggle-map-style"
              >
                <button
                  className="streets-button"
                  onClick={[Function]}
                />
              </div>,
              <i
                className="fas fa-bars streets"
                onClick={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="streets-button"
                  onClick={[Function]}
                />,
                "className": "toggle-map-style",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "streets-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-bars streets",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeMap": [Function],
            "displaySidebar": [Function],
            "login": [Function],
            "mapType": "streets",
            "redirect": false,
            "stateSidebar": false,
            "user": Object {
              "name": "paul",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div>
            <div
              className="toggle-map-style"
            >
              <button
                className="streets-button"
                onClick={[Function]}
              />
            </div>
            <i
              className="fas fa-bars streets"
              onClick={[Function]}
            />
          </div>
          <Routes
            changeMap={[Function]}
            displaySidebar={[Function]}
            login={[Function]}
            mapType="streets"
            redirect={false}
            stateSidebar={false}
            user={
              Object {
                "name": "paul",
              }
            }
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div
                className="toggle-map-style"
              >
                <button
                  className="streets-button"
                  onClick={[Function]}
                />
              </div>
              <i
                className="fas fa-bars streets"
                onClick={[Function]}
              />
            </div>,
            false,
            <Routes
              changeMap={[Function]}
              displaySidebar={[Function]}
              login={[Function]}
              mapType="streets"
              redirect={false}
              stateSidebar={false}
              user={
                Object {
                  "name": "paul",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="toggle-map-style"
                >
                  <button
                    className="streets-button"
                    onClick={[Function]}
                  />
                </div>,
                <i
                  className="fas fa-bars streets"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="streets-button"
                    onClick={[Function]}
                  />,
                  "className": "toggle-map-style",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "streets-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-bars streets",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeMap": [Function],
              "displaySidebar": [Function],
              "login": [Function],
              "mapType": "streets",
              "redirect": false,
              "stateSidebar": false,
              "user": Object {
                "name": "paul",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
